compiler/rustc_privacy/src/lib.rs

```diff
@@ -195,7 +195,7 @@ where
                 // Something like `fn() -> Priv {my_func}` is considered a private type even if
                 // `my_func` is public, so we need to visit signatures.
                 if let ty::FnDef(..) = ty.kind() {
-                    tcx.fn_sig(def_id).visit_with(self)?;
+                    tcx.fn_sig(def_id).subst_identity().visit_with(self)?;
```

Comment:
this should use `subst` imo but would impact stable code. Please add that as a `FIXME` and potentially fix that in a separate PR adding tests yourself.

When do we even hit this code. You shouldn't be able to name function definitions in the public signature. It would be fun to try and ICE for `FnDef` `Closure` and `Generator` to see where that breaks. it definitely will with `feature(generic_const_exprs)` :shrug: